src/Features/CSharp/Portable/EditAndContinue/CSharpEditAndContinueAnalyzer.cs

```diff
@@ -3206,6 +3244,10 @@ private static CheckedStatementSyntax TryGetCheckedStatementAncestor(SyntaxNode
             ReportUnmatchedStatements<ForEachStatementSyntax>(diagnostics, match, (int)SyntaxKind.ForEachStatement, oldActiveStatement, newActiveStatement,
                 areEquivalent: AreEquivalentActiveStatements,
                 areSimilar: (n1, n2) => SyntaxFactory.AreEquivalent(n1.Identifier, n2.Identifier));
+
+            ReportUnmatchedStatements<ForEachVariableStatementSyntax>(diagnostics, match, (int)SyntaxKind.ForEachVariableStatement, oldActiveStatement, newActiveStatement,
```

Comment:
Thank you! Done. Extended some methods with replacing SyntaxKind parameter by Func<SyntaxNode, bool> (predicate) parameter. Please let me know if we should extend it other way.

---
In reply to: [114644314](https://github.com/dotnet/roslyn/pull/19114#discussion_r114644314) [](ancestors = 114644314)